pkg/test/expectations/expectations.go

```diff
@@ -461,14 +463,33 @@ func ExpectOwnerReferenceExists(obj, owner client.Object) metav1.OwnerReference
 	return or
 }
 
+func ExpectFullyQualifiedNameFromCollector(collector prometheus.Collector) string {
```

Comment:
```suggestion
func ExpectMetricName(c prometheus.Collector) string {
```

Someone who is calling this probably doesn't care about the detail at that is the fully qualified name or is coming from a collector. I'd just simplify the name down to "This gets the metric name" and accepts a collector in as input. 

Can we also put some commenting above this function for what it's doing and why it's doing it? I spent a bit of time reading through this function to understand what we were doing and took a bit of time reading through the prometheus code to understand this limitation. Also, is it worth to go to the prometheus client_go folks and open an issue asking them if there's a way to make this less hard

Reply:
I can definitely add some additional comments, agreed that this function is not super straightforward. I actually originally named this `ExpectMetricName`, I changed it because the fully qualified name != the individual metric name although it does for our use cases. The complexity here is largely because I wanted a common function that would across individual metrics as well as vecs which required using a higher level interface.